ui.tests/test-module/specs/panelcontainer/panelcontainer.runtime.complex.spec.js

```diff
@@ -184,43 +185,82 @@ describe( "Form Runtime with Panel Container complex repeatability use cases ",
+            }).then(() => {
+                const allFields = formContainer.getAllFields();
+                const innerInstanceManager = allFields[innerInstanceManagerModel.id];
+                checkInstanceHTML(innerInstanceManager, 2)
+                    .then(() => {
+                        checkAddRemoveInstance(innerInstanceManager, 3, true)
+                            .then(() => {
+                                checkAddRemoveInstance(innerInstanceManager, 2)
+                                    .then(() => {
+                                        //min is 2, can't go below this
+                                        checkAddRemoveInstance(innerInstanceManager, 2);
+                                    });
+                            });
+                    });
             });
         });
 
     };
 
     const checkInnerInstance = (instancesModel, index) => {
-        const innerModelId = instancesModel.id;
-        const numberInputId = instancesModel[index].items[0].id;
-        const checkBoxGroupId = instancesModel[index].items[1].id;
+        const allFields = formContainer.getAllFields();
+        const instanceModel = instancesModel[index];
+        const innerModelId = instanceModel.id;
+        const innerView=allFields[innerModelId];
+        const numberInputId = instanceModel.items[0].id;
+        const numberInputView = allFields[numberInputId];
+        const checkBoxGroupId = instanceModel.items[1].id;
+        const checkBoxGroupView = allFields[checkBoxGroupId];
+        const innerLabelClass = innerView.getLabel().className;
+        const innerTooltipClass = innerView.getTooltipDiv().className;
+        const innerDescriptionClass = innerView.getDescription().className;
+        const numberInputLabelClass = numberInputView.getLabel().className;
+        const numberInputErrorClass = numberInputView.getErrorDiv().className;
+        const checkBoxGroupLabelClass = checkBoxGroupView.getLabel().className;
+        const checkBoxGroupViewErrorClass = checkBoxGroupView.getErrorDiv().className;
+
         cy.get(`#${innerModelId}`).should('exist');
-        cy.get(`#${innerModelId}-label`).should('exist');
-        cy.get(`#${innerModelId}-label`).invoke('attr', 'for').should('eq', innerModelId);
-        cy.get(`#${innerModelId}-shortDescription`).should('exist');
-        cy.get(`#${innerModelId}-longDescription`).should('exist');
-        cy.get(`#${numberInputId}-label`).should('exist');
-        cy.get(`#${numberInputId}-label`).invoke('attr', 'for').should('eq', numberInputId);
-        cy.get(`#${numberInputId}-errorMessage`).should('exist');
-        cy.get(`#${checkBoxGroupId}-label`).should('exist');
-        cy.get(`#${checkBoxGroupId}-label`).invoke('attr', 'for').should('eq', checkBoxGroupId);
-        return cy.get(`#${checkBoxGroupId}-errorMessage`).should('exist');
+        cy.get(`.${innerLabelClass}`).should('exist');
```

Comment:
this will give multiple results, you need to search within component, since its no longer id based.